### PR TITLE
Hauntium is chill with the undead, buffs hauntium stats

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Figured it'd be really cool if hauntium could be used by the undead

## Why It's Good For The Game

Right now it's pretty useless beyond some pretty good memes.

Should be pretty cool having weapons and armor that you can equip but will turn upon your foes if they use it!

## Changelog

:cl:
add: Hauntium can now be worn by the undead!
/:cl:

### CODE REVIEW

File: code/datums/ai/hauntium/haunted_controller.dm [hunk 1]
```diff
@@ -21,13 +24,22 @@
...
 	UnregisterSignal(pawn, COMSIG_ITEM_EQUIPPED)
-	var/list/hauntee_list = blackboard[BB_TO_HAUNT_LIST]
-	hauntee_list[equipper] = hauntee_list[equipper] + HAUNTED_ITEM_AGGRO_ADDITION //You have now become one of the victims of the HAAAAUNTTIIIINNGGG OOOOOO~~~
+	var/haunt_equipper = TRUE
+	if(isliving(equipper))
+		var/mob/living/possibly_cool = equipper
+		if(possibly_cool.mob_biotypes & MOB_UNDEAD)
+			haunt_equipper = FALSE
+	if(!haunt_equipper)
+		blackboard[BB_LIKES_EQUIPPER] = TRUE
```
Comment (tralezab) [inline code/datums/ai/hauntium/haunted_controller.dm:33]: possibly better to turn the AI off

File: code/datums/materials/hauntium.dm [hunk 2]
```diff
@@ -8,9 +8,9 @@
 	sheet_type = /obj/item/stack/sheet/hauntium
 	value_per_unit = 0.05
 	beauty_modifier = 0.25
-	strength_modifier = 1
-	armor_modifiers = list(MELEE = 1, BULLET = 1, LASER = 1, ENERGY = 1, BOMB = 1, BIO = 1, RAD = 1, FIRE = 1, ACID = 1)
-
+	//pretty good but only the undead can actually make use of these modifiers
+	strength_modifier = 1.2
+	armor_modifiers = list(MELEE = 1.1, BULLET = 1.1, LASER = 1.15, ENERGY = 1.15, BOMB = 1, BIO = 1, RAD = 1, FIRE = 1, ACID = 0.7)
```
Comment (tralezab) [inline code/datums/materials/hauntium.dm:13]: armor and strength is AROUND gold level with a few vulnerability swaps (acid weakness).

File: code/datums/ai/hauntium/hauntium_subtrees.dm [hunk 3]
```diff
@@ -2,6 +2,8 @@
 	var/obj/item/item_pawn = controller.pawn
 
 	if(ismob(item_pawn.loc)) //We're being held, maybe escape?
+		if(controller.blackboard[BB_LIKES_EQUIPPER])//don't unequip from people it's okay with
+			return SUBTREE_RETURN_FINISH_PLANNING
```
Comment (AMonkeyThatCodes) [inline code/datums/ai/hauntium/hauntium_subtrees.dm:6]: dont tell it to finish planning if this plan is not actually doing anything. If you finish planning it should finish with a reason.

Comment (tralezab) [reply code/datums/ai/hauntium/hauntium_subtrees.dm:6]: Shouldn't it be inert?

**Pull request merged**